android/src/main/java/com/appgoalz/rnjwplayer/RNJWPlayerView.java

```diff
@@ -290,62 +297,121 @@ public void setupPlayerView() {
+            // Destroy the player's rendering surface, we need to do this to prevent Android's
+            // MediaDecoders from crashing.
+            mPlayer.destroySurface();
 
-                }
+            mPlayerContainer = (ViewGroup) mPlayer.getParent();
 
-                @Override
-                public void onAllowRotationChanged(boolean b) {
-                    Log.e(TAG, "onAllowRotationChanged: "+b );
-                }
+            // Remove the JWPlayerView from the list item.
+            if (mPlayerContainer != null) {
+                mPlayerContainer.removeView(mPlayer);
+            }
+
+            // Initialize a new rendering surface.
+            mPlayer.initializeSurface();
 
+            // Add the JWPlayerView to the RootView as soon as the UI thread is ready.
+            mRootView.post(new Runnable() {
                 @Override
-                public void updateLayoutParams(ViewGroup.LayoutParams layoutParams) {
-//        View.setSystemUiVisibility(int).
-//        Log.e(TAG, "updateLayoutParams: "+layoutParams );
+                public void run() {
+                    mRootView.addView(mPlayer, new ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT,
+                        ViewGroup.LayoutParams.MATCH_PARENT
+                    ));
+                    mFullscreenPlayer = mPlayer;
                 }
+            });
+        }
 
+        @Override
+        public void onFullscreenExitRequested() {
+            mDecorView.setSystemUiVisibility(
+                View.SYSTEM_UI_FLAG_VISIBLE // clear the hide system flags
+            );
+
+            // Enter portrait mode
+            mActivity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
+            // Destroy the surface that is used for video output, we need to do this before
+            // we can detach the JWPlayerView from a ViewGroup.
+            mPlayer.destroySurface();
+
+            // Remove the player view from the root ViewGroup.
+            mRootView.removeView(mPlayer);
+
+            // After we've detached the JWPlayerView we can safely reinitialize the surface.
+            mPlayer.initializeSurface();
+
+            // As soon as the UI thread has finished processing the current message queue it
+            // should add the JWPlayerView back to the list item.
+            mPlayerContainer.post(new Runnable() {
                 @Override
-                public void setUseFullscreenLayoutFlags(boolean b) {
-//        View.setSystemUiVisibility(int).
-//        Log.e(TAG, "setUseFullscreenLayoutFlags: "+b );
+                public void run() {
+                    mPlayerContainer.addView(mPlayer, new ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT,
+                        ViewGroup.LayoutParams.MATCH_PARENT
+                    ));
+                    mFullscreenPlayer = null;
                 }
             });
```

Comment:
```suggestion
            });
            
            WritableMap eventExitFullscreen = Arguments.createMap();
            eventExitFullscreen.putString("message", "onFullscreenExit");
            getReactContext().getJSModule(RCTEventEmitter.class).receiveEvent(
                    getId(),
                    "topFullScreenExit",
                    eventExitFullscreen);

```

don't forget to add the `onFullScreen` props bridging